course_attendance_service/course_attendance_service/workflows/zoom_attendance_repo.py

```diff
@@ -0,0 +1,31 @@
+from typing import List
+
+import numpy as np
+
+from .attendance_workflows import Attendee
+from .zoom_attendance_api import ZoomAttendanceApi
+from .attendance_workflows import AttendanceRepo
+
+# Implementation of AttendanceRepo
+# RESPONSIBLE for AttendanceRepo contract
+# For Zoom attendance
+class ZoomAttendeeRepo(AttendanceRepo):
+    def __init__(self, zoom_api: ZoomAttendanceApi) -> None:
+        self.api = zoom_api
+        self.access_token = 'ACCESS TOKEN'
+
+    def list_all_attendees(self,workshop_id:str) -> List[str]:
+        report = self.api.get_zoom_participant_report(access_token=self.access_token,meeting_id=workshop_id)
+        participants = report['participants']
+        attendees = []
+        max_duration = np.nanmax([participant['duration'] for participant in participants if participant['status'] == 'in_meeting'])
+        attendance = False
```

Comment:
Deciding how a participant should be judged as "attended" or "not attended" for a given session is a higher-level policy.  Would it be possible to move this logic up to the Workflow level?

Reply:
- Moved logic to Workflow
- Modified Attendee class to accept duration
- attendance and duration are set to None by default and can be set dynamically